### PR TITLE
feat(`campaign`): add special allocations

### PR DESCRIPTION
**Pull request by lubtd**
Implementing special allocations structure and add it to campaign

- Type `SpecialAllocations`
- Add tests for type
- Add empty by default special allocations to campaign
- Update `CampaignShares` invariant to include special allocations

### CODE REVIEW

File: x/campaign/types/campaign_test.go [hunk 1]
```diff
@@ -45,6 +45,17 @@ func TestCampaign_Validate(t *testing.T) {
...
+		sample.Shares(r),
+		campaign.Shares(sdk.NewCoins(
+			sdk.NewCoin("foo", sdk.NewInt(100)),
+			sdk.NewCoin("s/bar", sdk.NewInt(200)),
+		)),
+	)
+	require.Error(t, invalidSpecialAllocations.Validate())
+	campaignInvalidSpecialAllocations := sample.Campaign(r, 0)
+	campaignInvalidSpecialAllocations.SpecialAllocations = invalidSpecialAllocations
+
```
Comment (aljo242) [inline x/campaign/types/campaign_test.go:58]: I think we should add a test where we are validating a "non-empty" special allocations field in the `campaign` object.

The best way to do this is to probably modify `sample.Campaign()` to automatically make that field non-null but valid

**Review by aljo242 — CHANGES_REQUESTED** [review]

File: x/campaign/types/campaign.go [hunk 2]
```diff
@@ -1,10 +1,10 @@
 package types
 
 import (
-	"errors"
 	"fmt"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/pkg/errors"
```
Comment (giunatale) [inline x/campaign/types/campaign.go:7]: was this dependency substitution intended?

**Review by giunatale — APPROVED** [review]
I am pre-approving, pending answer to my question above. Ping me if I need to re-review this

File: x/campaign/types/campaign.go (hunk 2 above)
Comment (lubtd) [reply x/campaign/types/campaign.go:7]: `github.com/pkg/errors` provides the `errors.Wrap` utility

**Comment by giunatale** [comment]
Unit tests are failing, it's not due to the usual port-binding issue.

**Comment by lubtd** [comment]
Simulation tests failing now with something we never had🤔

**Review by aljo242 — APPROVED** [review]
